### ISSUE TITLE
💡 Feature request: Ability for the instance to redirect reddit link into the instance link

### ISSUE THREAD

**Issue by openwick** [spikecodes/libreddit#557]
**TLDR** : libreddit instance itself should redirect reddit links by itself if provided to it either in search bar or url bar like [farside.link](https://github.com/benbusby/farside) or [invidious](https://invidio.us)

## Is your feature request related to a problem? Please describe.

Extensions like privacy redirect and libredirect can be used to redirect reddit links into libreddit links but  this option is not available in the mobile browsers so the instance should redirect the reddit link into libreddit link.

For example `https:libredd.it`  should take reddit url  say `https://www.reddit.com/r/aww/` either in search or as an url extension and return `https://libredd.it/r/aww/`

## Describe the feature you would like to be implemented

Redirecting from the url or search are the two options I am able to think

+ Redirecting from url : `https:libredd.it/https://www.reddit.com/r/aww/` should redirect to `https://libredd.it/r/aww/`   like another project called [farside.link](https://github.com/benbusby/farside) 
+ Redirecting from search : May be reddit url can be pasted in instance search and then it should automatically be redirected to instance link . For example invidous search can take youtube url in its search  and gives the video suggestion . If libreddit can do it , it solves the redirect issue.

## Describe alternatives you've considered

Another project called [farside.link](https://github.com/benbusby/farside) is doing exactly this but the problem is we cant choose the instance and the instance should be able to do this in the first place.

Extensions [libredirect](https://github.com/libredirect/libredirect) can automatically redirect the link but these extensions increase the browser fingerprint and not even available on mobile unless you install community forks of firefox and maually create a extension list which itself is a huge task and this option is not even available on chromium browsers. (Unless you use kiwi which is not secure )

## Additional context / screenshot

So in conclusion, I think the instance should have the redirect option for the urls rather than relying on other projects which increases finger printing and reduces security.

PS : If implemented please considered cases like `old.reddit.com` as well as reddit short links for redirection.
